docs/how-to-guides/creating-grc20.md

```diff
@@ -3,137 +3,142 @@ id: creating-grc20
-- Defined and set the value of local variable `admin` to point to a specific gno.land address of type `std.Address`.
-- Set the value of `mytoken` (type `*AdminToken`) to equal the result of creating a new token and configuring its name, symbol + decimal representation.
-- Minted 1 million `Mytoken` and set the administrator as the owner of these tokens.
+- Defined a new local variable `mytoken` and assigned that the type of
+pointer to `grc20.AdminToken`,
+- Defined and set the value of local variable `admin` to point to a specific
+address of type `std.Address`,
+- Initialize `mytoken` as a new GRC20 token, and set its name, symbol, and 
+decimal values,
+- Minted 1 million units of `My Token` and to the admin's address.
 
 ## 2. Adding token functionality
 
-The following section will be about introducing Public functions to expose functionality imported from the [grc20 package](https://github.com/gnolang/gno/tree/master/examples/gno.land/p/demo/grc/grc20).
-
-Add this imports :
-
-```go
-import (
-	....
-	"strings"
-
-	"gno.land/p/demo/ufmt"
-	"gno.land/r/demo/users"
-
-	pusers "gno.land/p/demo/users"
-)
-```
+In order to call exported functions from the `grc20` package, we also need to 
+expose them in the Realm. 
 
 [embedmd]:# (../assets/how-to-guides/creating-grc20/mytoken-2.gno go)
 ```go
 func TotalSupply() uint64 {
 	return mytoken.TotalSupply()
 }
 
-func BalanceOf(owner pusers.AddressOrName) uint64 {
-	balance, err := mytoken.BalanceOf(users.Resolve(owner))
+func Decimals() uint {
+	return mytoken.GetDecimals()
+}
+
+func BalanceOf(account std.Address) uint64 {
+	balance, err := mytoken.BalanceOf(account)
 	if err != nil {
 		panic(err)
 	}
+
 	return balance
 }
 
-func Allowance(owner, spender pusers.AddressOrName) uint64 {
-	allowance, err := mytoken.Allowance(users.Resolve(owner), users.Resolve(spender))
+func Allowance(owner, spender std.Address) uint64 {
+	allowance, err := mytoken.Allowance(owner, spender)
 	if err != nil {
 		panic(err)
 	}
+
 	return allowance
 }
 
-func Transfer(to pusers.AddressOrName, amount uint64) {
+func Transfer(recipient std.Address, amount uint64) {
 	caller := std.PrevRealm().Addr()
-	err := mytoken.Transfer(caller, users.Resolve(to), amount)
-	if err != nil {
+	if err := mytoken.Transfer(caller, recipient, amount); err != nil {
 		panic(err)
 	}
 }
 
-func Approve(spender pusers.AddressOrName, amount uint64) {
+func Approve(spender std.Address, amount uint64) {
 	caller := std.PrevRealm().Addr()
-	err := mytoken.Approve(caller, users.Resolve(spender), amount)
-	if err != nil {
+	if err := mytoken.Approve(caller, spender, amount); err != nil {
 		panic(err)
 	}
 }
 
-func TransferFrom(from, to pusers.AddressOrName, amount uint64) {
+func TransferFrom(from, to std.Address, amount uint64) {
 	caller := std.PrevRealm().Addr()
-	err := mytoken.TransferFrom(caller, users.Resolve(from), users.Resolve(to), amount)
-	if err != nil {
+	if err := mytoken.TransferFrom(caller, from, to, amount); err != nil {
```

Comment:
Should there be a check for the balance being positive and greater than 0?

```
if amount <= 0 {
        panic("Transfer amount must be greater than zero")
    }
 ```
 
 For mint and burn too:

```
if amount <= 0 {
        panic("Mint amount must be greater than zero")
    }
```

```
if amount <= 0 {
        panic("Burn amount must be greater than zero")
    }
```

Reply:
This does make sense, but looking through some ERC20 contracts, it seems no one is doing this check. I'll add it on our end.